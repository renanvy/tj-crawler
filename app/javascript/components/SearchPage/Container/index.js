import React from 'react'

export default function Container(props) {
  return (
    <div className="card">
      <div className="card-content">
        <span className="card-title">Processo n. 234354.343 do TJAL</span>

        <span>Distribuido em 09/11/19</span>
        
        <div className="row">
          {props.children}
        </div>
      </div>
    </div>
  )
}