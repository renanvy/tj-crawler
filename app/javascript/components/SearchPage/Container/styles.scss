.Container {
  &_title {
    margin: 0;
    font-size: 22px;
    font-weight: 500;
  }

  &_crawler_updated_at {
    display: flex;
    font-size: 16px;
    color: #7c7c7c;
    margin-top: 10px;
  }
}