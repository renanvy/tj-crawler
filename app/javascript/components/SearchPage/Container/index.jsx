import React, { Component } from "react";
import moment from "moment";
import PropTypes from "prop-types";

import "./styles.scss";

import Preloader from "../../common/Preloader";

export default class Container extends Component {
  renderContent() {
    const { process, loading, children } = this.props;

    if (loading) {
      return <Preloader />;
    }

    if (!process) {
      return <p className="center">Nenhum processo encontrado.</p>;
    }

    return (
      <>
        <h1 className="Container_title">
          Processo nº.
          {process.number}
          do TJAL
        </h1>

        <span className="Container_crawler_updated_at">
          Última atualização em          {" "}
          {moment(process.crawler_updated_at).format("DD/MM/YYYY HH:mm")}
        </span>

        <div className="row">{children}</div>
      </>
    );
  }

  render() {
    return (
      <div className="card">
        <div className="card-content">{this.renderContent()}</div>
      </div>
    );
  }
}

Container.propTypes = {
  process: PropTypes.object,
  loading: PropTypes.bool.isRequired,
  children: PropTypes.node
};
