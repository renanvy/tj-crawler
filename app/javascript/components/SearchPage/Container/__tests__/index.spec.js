import React from "react";
import renderer from "react-test-renderer";
import Container from "../index";

describe("Container", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <Container loading search={() => {}}>
        Foo
      </Container>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
