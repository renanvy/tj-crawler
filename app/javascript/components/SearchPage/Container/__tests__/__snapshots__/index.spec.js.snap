// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshot renders 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="Preloader"
    >
      <div
        className="preloader-wrapper big active"
      >
        <div
          className="spinner-layer spinner-blue-only"
        >
          <div
            className="circle-clipper left"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="gap-patch"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
